### ISSUE TITLE
HTTP header is prepended to file when uploading via public link

### ISSUE THREAD

**Issue by joernschellhaas** [owncloud/core#28745]
### Steps to reproduce
1. Create a public link with upload enabled
2. Let someone upload a file

### Expected behaviour
The file is saved to the server

### Actual behaviour
The file is saved to the server, however, the following is prepended:

```
-----------------------------7e12e2220a08
Content-Disposition: form-data; name="files[]"; filename="010.MOV"
Content-Type: video/quicktime
```

After removing this header, the file plays back fine.

### Server configuration
**Operating system**: Debian 3.16.43-2+deb8u2 (2017-06-26) x86_64 GNU/Linux

**Web server:** apache2 2.4.10-10+deb8u10

**Database:** default-mysql-server 1.0.1~bpo8+1

**PHP version:** php5 5.6.30+dfsg-0+deb8u1

**ownCloud version:** 10.0.0

**Updated from an older ownCloud or fresh install:** updated

**Where did you install ownCloud from:** Debian package

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

Can be provided in a gist if required

**List of activated apps:**

```
Enabled:
  - activity: 2.3.3
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.2.8
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.7.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 15.0.0
  - market: 0.1.0
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - example-theme
  - external
  - files_antivirus
  - user_external
```

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** "Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko"

**Operating system:** Windows

#### Web server error log
No corresponding entries

#### ownCloud log (data/owncloud.log)
````
{"reqId":"KlD7z8refxm2nUJdIOm6","level":3,"time":"2017-08-17T10:04:33+00:00","remoteAddr":"78.94.71.106","user":"--","app":"no app in context","method":"PUT","url":"\/public.php\/webdav\/010.MOV","message":"Exception: {\"Exception\":\"OC\\\\User\\\\NoUserException\",\"Message\":\"Attempted to initialize mount points for null user and no user in session\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(212): OC\\\\Files\\\\Filesystem::initMountPoints(false)\\n#1 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(150): OCA\\\\Activity\\\\FilesHooks->getSourcePathAndOwner('\\\/Syncthing\\\/bild...')\\n#2 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/FilesHooks.php(106): OCA\\\\Activity\\\\FilesHooks->addNotificationsForFileAction('\\\/Syncthing\\\/bild...', 'file_created', 'created_self', 'created_by')\\n#3 \\\/var\\\/www\\\/owncloud\\\/apps\\\/activity\\\/lib\\\/FilesHooksStatic.php(45): OCA\\\\Activity\\\\FilesHooks->fileCreate('\\\/Syncthing\\\/bild...')\\n#4 [internal function]: OCA\\\\Activity\\\\FilesHooksStatic::fileCreate(Array)\\n#5 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/legacy\\\/hook.php(105): call_user_func(Array, Array)\\n#6 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(292): OC_Hook::emit('OC_Filesystem', 'post_create', Array)\\n#7 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/File.php(221): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->emitPostHooks(false)\\n#8 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/Sabre\\\/Directory.php(158): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\File->put(Resource id #72)\\n#9 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(1095): OCA\\\\DAV\\\\Connector\\\\Sabre\\\\Directory->createFile('010.MOV', Resource id #72)\\n#10 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/CorePlugin.php(525): Sabre\\\\DAV\\\\Server->createFile('010.MOV', Resource id #72, NULL)\\n#11 [internal function]: Sabre\\\\DAV\\\\CorePlugin->httpPut(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\\n#13 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(479): Sabre\\\\Event\\\\EventEmitter->emit('method:PUT', Array)\\n#14 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(254): Sabre\\\\DAV\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#15 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/publicwebdav.php(90): Sabre\\\\DAV\\\\Server->exec()\\n#16 \\\/var\\\/www\\\/owncloud\\\/public.php(76): require_once('\\\/var\\\/www\\\/ownclo...')\\n#17 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/Files\\\/Filesystem.php\",\"Line\":399}"}````

**Comment by kdslkdsaldsal** [comment]
> ownCloud version: 10.0.0

Could you retry with a recent versions as asked in the issue template you had filled out:

> Before reporting any issues please make sure that you're using the latest available version for your major branch (e.g. 9.0.x), see https://owncloud.org/changelog/

-> https://owncloud.org/changelog/#latest10.0

Something similar was already fixed in 10.0.1 with https://github.com/owncloud/core/pull/27875

**Comment by joernschellhaas** [comment]
I'm sorry, I had configured the package sources in a way that it would only upgrade to 10.0.0.

Although I cannot test the new version because it was uploaded from a third party, I think the problem is solved by the changes you mentioned. Thanks for the support!
